src/main/java/com/sourcegraph/vcs/RepoUtil.java

```diff
@@ -128,7 +132,25 @@ private static String doReplacements(
 
     if (vcsType == VCSType.GIT && repository != null) {
       String cloneURL = GitUtil.getRemoteRepoUrl((GitRepository) repository, project);
-      return convertGitCloneURLToCodebaseNameOrError(cloneURL).getValue();
+      AtomicReference<String> repoURl = new AtomicReference<>();
```

Comment:
This isn't right. `withAgent` runs the callback on a separate thread, so you will often--but not always--have null in `repoURl`.

You need to:

- Make `getRemoteRepoUrl` return a CompletableFuture<String?>
- Convert the callers to handle an asynchronous result. This may take care and attention to not block EDT, not starve a threadpool, ensure that if you started on EDT you continue on EDT if the caller requires EDT, etc.
- Convert the method body to handle the asynchronous result: Create the future, run the code with Agent, complete the result in the callback.

By the way, let's pay attention to details like `repoURL` or `repoUrl` but not `repoURl`

Reply:
Ah for some reason I thought it was blocking. I've updated it in [this commit](https://github.com/sourcegraph/jetbrains/pull/2257/commits/a40f65eb9dcedeb28addc9f3b2e3cdbe1ca1ba9a?diff=split&w=0). Do you know if there are any unit tests of agent integration? Would this be a good use case for an integration test?